src/Controls/src/Core/Page/Page.cs

```diff
@@ -852,6 +852,22 @@ internal void SendNavigatedFrom(NavigatedFromEventArgs args, bool disconnectHand
 			if (args.NavigationType == NavigationType.Pop ||
 				args.NavigationType == NavigationType.PopToRoot)
 			{
+				// Don't dispose Handlers too early on iOS!
+				// iOS aggressively cleans up page Handlers during navigation, but if the page
```

Comment:
This is only required on iOS? Because this is a cross platform code that will run same on other platforms.